test/types/hooks.test-d.ts

```diff
@@ -84,18 +88,18 @@ server.addHook('onRoute', (opts) => {
 server.addHook('onRegister', (instance, done) => {
   expectType<FastifyInstance>(instance)
   expectType<(err?: FastifyError) => void>(done)
-  expectType<void>(done(new Error()))
+  expectType<void>(done(new FastifyErrorImpl()))
 })
 
 server.addHook('onReady', function (done) {
   expectType<(err?: FastifyError) => void>(done)
-  expectType<void>(done(new Error()))
+  expectType<void>(done(new FastifyErrorImpl()))
 })
 
 server.addHook('onClose', (instance, done) => {
   expectType<FastifyInstance>(instance)
   expectType<(err?: FastifyError) => void>(done)
-  expectType<void>(done(new Error()))
+  expectType<void>(done(new FastifyErrorImpl()))
```

Comment:
It's it requires that we pass a FastifyError here? The `code` property should not be required for this